### DISCUSSION TITLE
No ADB option, Android TV remote integration..

### DISCUSSION THREAD

**Discussion by Blueion76** [PRProd/HA-Firemote#310] [Ideas]
Hey, I greatly appreciate the work here you've done on Firemote. Is it possible to make a working remote using only the android tv remote integration in the card? My Onn box has ADB but since I'm using an Ethernet adapter it seems to disable the Ethernet adapter whenever I enable developer options. The Android TV remote integration works well and I don't have the network issues as I do with ADB on my Onn. Google TV Box. Thanks!

**Comment by PRProd** [comment]
Greetings @Blueion76!

You may have seen a similar question in the [FAQ](https://github.com/PRProd/HA-Firemote/wiki#faq) section of the [Firemote Wiki](https://github.com/PRProd/HA-Firemote/wiki) called [Why do I still need the ADB Integration when thre is an android remote integration?](https://github.com/PRProd/HA-Firemote/wiki#why-do-i-still-need-the-adb-integration-when-there-is-an-android-remote-integration)  If you haven't read that yet, I suggest taking a quick look.

As far as using an Onn. box goes, it's not yet an [officially supported Firemote device](https://github.com/PRProd/HA-Firemote/wiki#devices-officially-supported-by-firemote).  That being said, Onn. currently has the most votes over in the discussion [Poll: Vote for what gets added next! #41](https://github.com/PRProd/HA-Firemote/discussions/41).  If you haven't voted since the last poll reset on Jan-24-2024, I suggest placing your vote today.

Now, onto your real question: How to use ADB and a USB Ethernet adapter at the same time.  I bought [an adapter](https://www.amazon.com/gp/product/B09HGZ2XXQ?th=1) to use with my [Xiaomi Mi Box S](https://www.amazon.com/gp/product/B07KLWGGYS/?th=1) and I soon discovered this same issue you are describing.  I did find a way to make it work, but it can be a bit frustrating and it always stops working if the Mi Box loses power.  (BTW: How is your onn. device receiving power in this configuration?  Is there a POE switch behind your adapter?)

This is the not so fun procedure that I go through after a power cycle of my Mi Box / Android TV device:
 1. Open the Settings (gear) menu
 
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/a74b14c6-803e-4a66-857e-770cac212d57)

 2. Select Device Preferences
 
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/35b8936f-7a4f-47e7-8c6c-17b012c6b8fa)

 3. Select Developer Options
 
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/2ac30a7d-19c3-454b-902e-063f7b0de806)

 4. Scroll down and turn USB Debugging Off
 
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/92f7b32f-7e35-42c7-b1cd-fce3c75cecdb)

 5. Scroll down further to open the Select USB Configuration menu
 
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/6417293e-a0a3-4c13-8c73-bffd10588df3)

 6. Change to RNDIS (USB Ethernet)
 
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/b4114a48-ef83-4cc0-a25a-1c7e731fa13f)

 7. Hit the back button on your remote several times to back out of the settings menu and return to the home screen
 
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/bb748f43-cf27-4363-a285-cf15d7a3de5c)

 8. Re-Open the Settings menu
 
 ![image](https://github.com/PRProd/HA-Firemote/assets/25486599/a74b14c6-803e-4a66-857e-770cac212d57)
 
 9. Select Device Preferences
  
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/35b8936f-7a4f-47e7-8c6c-17b012c6b8fa)

 10. Select Developer Options
  
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/2ac30a7d-19c3-454b-902e-063f7b0de806)

 11. Turn USB Debugging On
 
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/f57f267c-9bee-4b53-9161-45e69012b923)


This procedure usually works for me.  Sometimes I find that I have to wait for a few seconds after backing out of the menus (step 7) while the ethernet connection is being re-established.

Hopefully this procedure is similar to one that might make things work in your situation.  Let me know!  I think I'll likely be picking up an onn. box pretty soon anyway.

-Doug

**Reply by Blueion76** [reply]
Appreciate the response, it usually only happens after enabling USB debugging and a reboot. I'll give this a try when I get home though. Currently I have a USB OTG splitter (power input, USB A port and micro USB into the Onn.) the Ethernet works flawlessly until I enable developer options and have a reboot which is unfortunate.

**Reply by Blueion76** [reply]
Adding on to my previous response, I have the Onn integrated by saying it's a CC w/ Google TV which seems to work fine. Everything functions well when ADB is working. I think just the remote UI needs to be integrated into FireMote.

**Reply by Blueion76** [reply]
> Appreciate the response, it usually only happens after enabling USB debugging and a reboot. I'll give this a try when I get home though. Currently I have a USB OTG splitter (power input, USB A port and micro USB into the Onn.) the Ethernet works flawlessly until I enable developer options and have a reboot which is unfortunate. 

Tried this yesterday and Ethernet connectivity still disappears after a reboot unfortunately. Correction on the enabling developer options part: it's only after I enable USB debugging and reboot - Ethernet connectivity disappears. Normally I'd say it's fine but this box seems to randomly reboot sometimes when not in use so it's not ideal to have to disable USB debugging reboot then reenable.

**Reply by PRProd** [reply]
Yeah, same.  I guess I could use wifi, but...?

**Comment by Swiiney** [comment]
Hi,
Same kind of problem with latest android version  on Chromecast. ADB integration does not work anymore with Android 14 and so I cannot use anymore the remote.
A no ADB option would be great, even if we loose some capabilities.

**Reply by PRProd** [reply]
I see that you've opened #704 which I'm just getting around to looking at right now.  (Sorry for the long delay, I had some things going on in my life that needed attention. Now I'm finally getting back to some fun Firemote things)

Since it's technically NOT a Firemote thing, BUT it DOES greatly hinder the ability of Firemote to easily handle Chromecast devices, I'm going to keep #704 open and discuss things there.
